tmt/steps/provision/beaker_mrack.py

```diff
@@ -0,0 +1,508 @@
+                            operator = chunk
+
+                return operator, amount
+
+            def _translate_tmt_hw(self, hw: Dict[str, Any]) -> Dict[str, Any]:
+                key = "_key"
+                value = "_value"
+                op = "_op"
+
+                system = {}
+                disks = []
+                cpu = {}
+
+                for key, val in hw.get("constraints", {}).items():
+                    if key == "memory":
+                        operator, amount = self._parse_amount(val)
+                        system.update({
+                            key: {
+                                value: amount,
+                                op: operator
+                                }
+                            })
+                    if key == "disk":
+                        for dsk in val:
+                            operator, disk = self._parse_amount(dsk["size"])
+                            disks.append({
+                                "disk": {
+                                    "size": {
+                                        value: disk,
+                                        op: operator,
+                                        }
+                                    }
+                                })
+                    if key == "cpu":
+                        if val.get("processors"):
+                            cpu.update({
+                                "cpu_count": {
+                                    value: val["processors"],
+                                    op: "=",
+                                    }
+                                })
+                        if val.get("model"):
+                            cpu.update({
+                                "model": {
+                                    value: val["model"],
+                                    op: "=",
+                                    }
+                                })
+
+                and_req = []
+                for rec in [system, disks, cpu]:
+                    if not rec:
+                        continue
+                    if isinstance(rec, dict):
+                        and_req.append(rec)
+                    if isinstance(rec, list):
+                        and_req += rec
+
+                host_req = {}
+                if and_req:
+                    host_req = {
+                        "hostRequires": {
+                            "and": and_req
+                            }
+                        }
+
+                return host_req
+
+            def create_host_requirement(self, host: Dict[str, Any]) -> Dict[str, Any]:
+                """Create single input for Beaker provisioner."""
+                mrack_req: Dict[str, Any] = self._translate_tmt_hw(host.get("hardware", {}))
+                host.update({"beaker": mrack_req})
+                req: Dict[str, Any] = super().create_host_requirement(host)
+                return req
+
+    except ImportError:
```

Comment:
I'd move this right after `import mrack` - takes care of *import* issues, there's no need to include so much code in the block when the import is already known to be successful.